paddle/fluid/operators/distributed/grpc_client.cc

```diff
@@ -87,10 +91,16 @@ VarHandlePtr GRPCClient::AsyncSendVar(const std::string& ep,
     // stub context
     s->response_call_back_ = nullptr;
 
+    platform::RecordEvent record_event(method, p_ctx);
+
     auto call = s->stub_g_.PrepareUnaryCall(
         s->context_.get(), "/sendrecv.SendRecvService/SendVariable", req, &cq_);
     call->StartCall();
     call->Finish(&s->reply_, &s->status_, reinterpret_cast<void*>(s));
+
+    if (platform::IsProfileEnabled()) {
```

Comment:
```suggestion
    if (UNLIKELY(platform::IsProfileEnabled())) {
```

to speed up normal case, so this change can not affect normal performance.

Reply:
Or maybe put it to `SendProcessor` so it will record the RPC lifetime and do not break the overall execution.